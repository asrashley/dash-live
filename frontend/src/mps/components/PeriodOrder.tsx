import { type JSX } from "preact";

import { MenuItemType } from "../../types/MenuItemType";
import { DropDownMenu } from "../../components/DropDownMenu";
import { Icon } from '../../components/Icon';

export interface PeriodOrderProps {
  addPeriod: (ev: JSX.TargetedEvent<HTMLAnchorElement>) => void;
  deletePeriod: (ev: JSX.TargetedEvent<HTMLAnchorElement>) => void;
}

export function PeriodOrder({ addPeriod, deletePeriod }: PeriodOrderProps) {
  const menu: MenuItemType[] = [
    {
      title: "Add another Period",
      onClick: addPeriod,
    },
    {
      title: "Delete Period",
      onClick: deletePeriod,
    },
  ];

  return <DropDownMenu linkClass="" menu={menu}>
    <Icon name="three-dots" />
  </DropDownMenu>;
}
