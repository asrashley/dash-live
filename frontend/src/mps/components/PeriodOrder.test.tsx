import { afterEach, describe, expect, test, vi } from "vitest";
import { renderWithProviders } from "../../test/renderWithProviders";
import { PeriodOrder } from "./PeriodOrder";
import { fireEvent } from "@testing-library/preact";

describe("PeriodOrder component", () => {
  const addPeriod = vi.fn();
  const deletePeriod = vi.fn();

  afterEach(() => {
    vi.clearAllMocks();
  });

  test("matches snapshot", () => {
    const { asFragment } = renderWithProviders(
      <PeriodOrder addPeriod={addPeriod} deletePeriod={deletePeriod} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('add a period', () => {
    const { getByText } = renderWithProviders(
        <PeriodOrder addPeriod={addPeriod} deletePeriod={deletePeriod} />
      );
      const elt = getByText('Add another Period') as HTMLElement;
      fireEvent.click(elt);
      expect(addPeriod).toHaveBeenCalledTimes(1);
      expect(deletePeriod).not.toHaveBeenCalled();
  });

  test('delete a period', () => {
    const { getByText } = renderWithProviders(
        <PeriodOrder addPeriod={addPeriod} deletePeriod={deletePeriod} />
      );
      const elt = getByText('Delete Period') as HTMLElement;
      fireEvent.click(elt);
      expect(addPeriod).not.toHaveBeenCalled();
      expect(deletePeriod).toHaveBeenCalledTimes(1);
  });
});
