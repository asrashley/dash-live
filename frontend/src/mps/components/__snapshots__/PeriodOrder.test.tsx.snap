// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PeriodOrder component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <a
      aria-expanded="false"
      class=" dropdown-toggle"
      href="#"
      role="button"
    >
      <span
        aria-label="three-dots icon"
        class="bi icon bi-three-dots "
        role="img"
      />
    </a>
    <ul
      class="dropdown-menu "
    >
      <li
        data-testid="ddi_0"
      >
        <a
          class="dropdown-item"
          href="#"
        >
          Add another Period
        </a>
      </li>
      <li
        data-testid="ddi_1"
      >
        <a
          class="dropdown-item"
          href="#"
        >
          Delete Period
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
